### ISSUE TITLE
Support arguments (-a) for dlopen

### ISSUE THREAD

**Issue by vimkim** [namhyung/uftrace#1981]
Is there a way to show arguments for functions inside a dynamically loaded libraries using 
dlopen?

### How to Reproduce

libexample.c
```c
#include <stdio.h>

void say_hello(int x) {
    printf("Hello from the shared library! x = %d\n", x);
}
```

main.c
```c
#include <stdio.h>
#include <dlfcn.h>

void say_hello_no_dlopen(int x){
  printf("Hello, World! x = %d\n", x);
  printf("This function is not a dynamically loaded function\n\n");
}

int main() {

    int y = 333;

    say_hello_no_dlopen(y);

    // Load the shared library
    void *handle = dlopen("./libexample.so", RTLD_LAZY);
    if (!handle) {
        fprintf(stderr, "Error: %s\n", dlerror());
        return 1;
    }

    // Clear any existing error
    dlerror();

    // Get the function pointer
    void (*say_hello)(int) = (void (*)(int))dlsym(handle, "say_hello");
    const char *error = dlerror();
    if (error) {
        fprintf(stderr, "Error: %s\n", error);
        dlclose(handle);
        return 1;
    }

    // Call the function
    say_hello(777);

    // Close the shared library
    dlclose(handle);

    return 0;
}


```

Compile and Run:
```sh
gcc -shared -finstrument-functions -fPIC -pg -g -o libexample.so libexample.c
gcc -finstrument-functions -g -pg -o main main.c -ldl
./main
```

Results:
```txt
Hello, World! x = 333
This function is not a dynamically loaded function

Hello from the shared library! x = 777
```

Uftrace:
```
uftrace -a ./main
```

Results:
![image](https://github.com/user-attachments/assets/20f32ac4-1de8-4758-85e4-0ed17bdf4393)

As you can see, 
the function named 'say_hello_no_dlopen()' prints the argument (of int 333) just fine,
while the dlopened function say_hello prints no argument.

Do you know how to fix this issue?

**Comment by honggyukim** [comment]
Hi @vimkim, this is duplicate with #842 so please have a look.

**Comment by namhyung** [comment]
Btw, please don't use `-finstrument-functions` and `-pg` together.

**Comment by namhyung** [comment]
Fixed by 2b9e0bb.
